ci-operator/step-registry/ibmcloud/provision/dedicatedhost/static/ibmcloud-provision-dedicatedhost-static-commands.sh

```diff
@@ -0,0 +1,143 @@
+set -o pipefail
+
+#pre create the dedicated hosts based on the ${SHARED_DIR}/dedicated_host.yaml and update "${SHARED_DIR}/dedicated_host.yaml
+#save the dedicated host name for master nodes in "${SHARED_DIR}/dedicated_host"
+#save the dedicated host name for worker nodes in "${SHARED_DIR}/dedicated_host" 
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+# IBM Cloud CLI login
+function ibmcloud_login {
+    export IBMCLOUD_CLI=ibmcloud
+    export IBMCLOUD_HOME=/output
+    region="${LEASED_RESOURCE}"
+    export region
+    echo "Try to login..."
+    "${IBMCLOUD_CLI}" login -r ${region} --apikey @"${CLUSTER_PROFILE_DIR}/ibmcloud-api-key"
+}
+
+## create the dedicated host group and dedicated host, if succeed, save the dedicated host name to file"
+## input parameter: the resource group of the dhg, dh profile, zone, name of the dh and the file which save the created dh name.
+function createDH() {
+    local rg=$1 dProfile=$2 zone=$3 dhName=$4 file=$5
+    local tmp family class dhg status
+    tmp=$(${IBMCLOUD_CLI} is dedicated-host-profile $dProfile --output JSON | jq -r '.family +","+ .class')
+    family=${tmp%,*}
+    class=${tmp#*,}
+    echo "family:" $family " class:" $class
+    dhg="${dhName}g"
+    run_command "${IBMCLOUD_CLI} is dedicated-host-group-create --zone $zone --family $family --class $class --name ${dhg}"
+    
+    ${IBMCLOUD_CLI} is dedicated-host-create --profile $dProfile --name $dhName --dhg $dhg --output JSON
+    
+    status=$(waitingStatus ${dhName})
+    echo "${dhName} status: ${status}"
+    run_command "${IBMCLOUD_CLI} is dedicated-host ${dhName}"
+
+    if [[ "${status}" != "stable" ]]; then
+        echo "ERROR: fail to create the dedicated host with profile $dProfile on zone: $zone resouce group:$rg"
+        echo "delete the dh ${dhName}, which status is ${status} ..."
+        run_command "${IBMCLOUD_CLI} is dhd ${dhName} -f"
+        run_command "${IBMCLOUD_CLI} is dhgd ${dhg} -f"
+    else
+        echo "${dhName}" > "${file}"        
+    fi
+}
+
+# waiting the pending dh status changed
+function waitingStatus() {
+    local dh=$1 status
+    COUNTER=0
+    while [ $COUNTER -lt 200 ]
+    do 
+        sleep 10
+        COUNTER=`expr $COUNTER + 20`
+        status=$(${IBMCLOUD_CLI} is dh $dh --output JSON | jq -r ."lifecycle_state")
+        if [[ "${status}" != "pending" ]]; then
+            break
+        fi
+    done
+    echo ${status}
+}
+
+#####################################
+##############Initialize#############
+#####################################
+
+dhProfile="bx2-host-152x608"
+masterProfile="bx2-8x32"
+workerProfile="bx2-4x16"
+
+echo "the dh profile of the master nodes: ${dhProfile}; the dh profile of the worker nodes ${dhProfile}"
+
+echo "$(date -u --rfc-3339=seconds) - Creating the Dedicated Host..."
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+
+dhName="${CLUSTER_NAME}-dh"
+
+dhgRGFile=${SHARED_DIR}/ibmcloud_resource_group
```

Comment:
vpc and bastion is using the same name of file- ${SHARED_DIR}/ibmcloud_resource_group, use to use a different name.

Reply:
ibmcloud_resource_group file records the rg which was created before install

Reply:
To avoid name conflicting in the future, pls change to another different name,

Reply:
if we need support vpc+dedicated host, the file name needs to change

Reply:
Maybe some day we need to support multiple resource group for one cluster install in some day, that is why I suggest to use a different name, I do not expect to run more code change in the future.  \